StRoot/StEvent/StFcsHit.h

```diff
@@ -100,9 +104,10 @@ class StFcsHit : public StObject {
     Float_t  mEnergy=0.0;     // corrected energy
     StFcsCluster* mCluster=0; // pointer to cluster this hit belongs
     TArrayS* mData=0;         // 12bit ADC values + flag at highest 4 bits, array of timebin
-    
-    ClassDef(StFcsHit,5)
 
+    vector<pair<unsigned int, float>> mGeantTracks; // parent G2T track id and dE
```

Comment:
I don't know how to use set<pair<unsigned int, float>> and later modify (add) to float in the pair.... Please let me know.

Reply:
If you change your `StFcsHit::addGeantTrack(unsigned int id, float e)` to work with iterators as suggested then it will work for the both `vector` and `set` types. Please note that I am not suggesting to switch to `set` of pairs right away. I have no idea how the performance may change because I don't know how many inserts or look-ups you expect. I have no real argument for a `set` other than I see your intention to store sorted and unique elements, hence, a `set` would be my first choice but a `vector` may work just as well.

In case you do decide to investigate the `set`, all you need to do is to redefine the default comparator for `std::pair<unsigned, float>` to something like this:

```c++
auto cmp = [](pair<unsigned, float> a, pair<unsigned, float> b) { return a.first < b.first; };

std::set< pair<unsigned, float>, decltype(cmp)> s(cmp);
```

Reply:
Isn't this calling for `std::map`?

Reply:
For the most part, the hits from a single track which initiates an EM or hadronic cascade will be presented to the slow simulator before the next track is processed.  So the vector is the natural choice here.  For any geant hit added to the FCS hit, the pair which needs to be updated is likely to be at the back of the vector.  Once all hits have been processed, we can loop over the FCS hits and ensure that they are sorted descending in energy.